crates/runtime/src/accelerated_table.rs

```diff
@@ -60,38 +71,71 @@ impl AcceleratedTable {
         refresh_interval: Option<Duration>,
         object_store: Option<(Url, Arc<dyn ObjectStore + 'static>)>,
     ) -> Self {
+        let (refresh_trigger, refresh_trigger_receiver) = mpsc::channel::<()>(1);
+
         let refresh_handle = tokio::spawn(Self::start_refresh(
             dataset_name,
             Arc::clone(&federated),
             refresh_mode,
             refresh_interval,
             Arc::clone(&accelerator),
             object_store,
+            refresh_trigger.clone(),
+            refresh_trigger_receiver,
         ));
 
         Self {
             accelerator,
             federated,
             refresh_handle,
+            refresh_trigger,
         }
     }
 
+    #[allow(dead_code)]
+    pub async fn trigger_refresh(&self) -> Result<()> {
+        self.refresh_trigger
+            .send(())
+            .await
+            .context(FailedToTriggerRefreshSnafu)?;
+        Ok(())
```

Comment:
In the new model I specified earlier, `refresh_trigger` would become an Option and this would return an error if it was `None` (i.e trying to refresh a dataset that is configured as Append)

Reply:
Updated - thank you